api/src/main/java/jakarta/data/metamodel/constraint/NullRecord.java

```diff
@@ -15,13 +15,13 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-package jakarta.data.metamodel.restrict;
+package jakarta.data.metamodel.constraint;
 
-public interface UnaryRestriction<T> extends Restriction<T> {
-    String attribute();
-
-    UnaryOperator operator();
+import jakarta.data.metamodel.restrict.Operator;
 
+record NullRecord<T>() implements Null<T> {
```

Comment:
equals and hashCode are missing
```suggestion
record NullRecord<T>() implements Null<T> {
    @Override
    public boolean equals(Object obj) {
        return obj instanceof NullRecord;
    }

    @Override
    public int hashCode() {
        return 0;
    }
```